src/main/java/org/jvnet/solaris/libzfs/ZFSObject.java

```diff
@@ -503,10 +506,54 @@ public int callback(zfs_handle_t handle, Pointer arg) {
     }
 
     /**
+     * Grants the specified set of permissions to this dataset.
+     * NOTE that the "zfs_perm_*" libzfs.so routines are not present at
+     * all in OpenZFS, nor even in Sun Solaris 10u8 (last seen in 10u6).
+     * Not sure about Sol10u7 since it was a rather intermediate release.
+     * They may have been replaced with other logic, since zfs_allow_t and
+     * related data types and comments are still present in the headers.
+     */
+    public void allow(ACLBuilder acl) {
+        String abi = library.getFeature("LIBZFS4J_ABI_zfs_perm_set");
+        if (abi.equals("pre-sol10u8")) {
+            for (PermissionBuilder b : acl.builders) {
+                if(LIBZFS.zfs_perm_set(handle,b.toNativeFormat(this))!=0)
+                    throw new ZFSException(library);
+            }
+        } else {
+            if (abi == null)
+                abi = "<null>";
+            LOGGER.log(Level.FINE, "libzfs4j::allow() was called while LIBZFS4J_ABI_zfs_perm_set=='"+abi+"'");
+        }
+
+    }
+
+    /**
+     * Revokes the specified set of permissions to this dataset.
+     * See also the detailed "NOTE" comment above.
+     */
+    public void unallow(ACLBuilder acl) {
```

Comment:
Should be fixed now (regarding logged "NO-OP").

IMHO should not be fixed regarding "deprecation", per comments above. OR perhaps it can be "deprecated" now, and "un-deprecated" when the `libzfs.jar` begins implementing these functions for new ZFS-enabled OSes. Do I gather correctly that marking the method as deprecated would let consumers see warnings during their compilation and/or run-time, so we know if we use these methods in e.g. Jenkins-related codebase?